js/test/date/testdatefmtasync.js

```diff
@@ -304,5 +304,166 @@ module.exports.testdatefmtasync = {
+            test.done();
+            return;
+        }
+        test.expect(2);
+        new DateFmt({
+            length: "short",
+            locale: "ko-KR",
+            useIntl: true,
+            sync: false,
+            onLoad: function(fmt) {
+                test.ok(fmt !== null);
+                var date = new Date(2022, 4, 29);
+                test.equal(fmt.format(date), "22. 5. 29.");
+                test.done();
+            }
+        });
+    },
+    testDateFmtIntlDateTimeObjAsync_ko_KR_Medium: function(test) {
+        if(!DateFmt.isIntlDateTimeAvailable("ko-KR")){
+            // The result is different depending on the node version.
+            test.done();
+            return;
+        }
+        test.expect(2);
+        new DateFmt({
+            length: "medium",
+            locale: "ko-KR",
+            useIntl: true,
+            sync: false,
+            onLoad: function(fmt) {
+                test.ok(fmt !== null);
+                var date = new Date(2022, 4, 29);
+                test.equal(fmt.format(date), "2022. 5. 29.");
+                test.done();
+            }
+        });
+    },
+    testDateFmtIntlDateTimeObjAsync_ko_KR_Long: function(test) {
+        if(!DateFmt.isIntlDateTimeAvailable("ko-KR")){
+            // The result is different depending on the node version.
+            test.done();
+            return;
+        }
+        test.expect(2);
+        new DateFmt({
+            length: "long",
+            locale: "ko-KR",
+            useIntl: true,
+            sync: false,
+            onLoad: function(fmt) {
+                test.ok(fmt !== null);
+                var date = new Date(2022, 4, 29);
+                test.equal(fmt.format(date), "2022년 5월 29일");
+                test.done();
+            }
+        });
+    },
+    testDateFmtIntlDateTimeObjAsync2_en_US: function(test) {
+        if(!DateFmt.isIntlDateTimeAvailable("en-US")){
+            // The result is different depending on the node version.
+            test.done();
+            return;
+        }
+        test.expect(2);
+        DateFactory({
+            year: 2022,
+            month: 9,
+            day: 29,
+            hout:13,
+            minute: 32,
+            sync: false,
+            onLoad:function(date){
+                test.ok(date !== null);
+                new DateFmt({
+                    length: "long",
+                    locale: "en-US",
+                    useIntl: true,
+                    sync: false,
+                    onLoad: function(fmt){
+                        if(ilib._getPlatform() === "nodejs"){
+                            var version = process.versions["node"];
+                            var majorVersion = version.split(".")[0];
+                            //console.log("version: " + version);
+                            if(majorVersion == "8" || majorVersion == "10"){
+                                test.equal(fmt.format(DateFactory._ilibToDate(date)), "9/29/2022");
```

Comment:
`format()` already calls `DateFactory._ilibToDate()` doesn't it?

Reply:
Okay just checked. It does. You should be able to do `fmt.format(date)`

Reply:
In this case, it returns Intl's object. so this format method is for Intl's one. not iLib one.  It should take js date, instead iLib date style.

Reply:
Oh wait, that's a problem. The "useIntl" parameter is a request to use Intl if it is available. If it is not available, or if the parameters cannot be mapped to Intl, or if the locale is not support, then the format object returned is the ilib object. If Intl is available, then it is the Intl object. The caller won't know until they check the object. Also, it is weird for a constructor to return an object of a different type than itself.

It would be better to change the code so that instead of returning the Intl formatter, you always return the ilib formatter:

```javascript
constructor:
if (useIntl) { this.intlFormatter = Intl.dateTimeFormat(); return; }

format:
if (useIntl) { convert date to JS date; return this.intlFormatter(JS Date); }
```

Reply:
I updated it. Please re-check it.